budgetportal/templatetags/define_action.py

```diff
@@ -1,12 +1,21 @@
 from django import template
+from django.utils.safestring import mark_safe
+
+import json
+
 register = template.Library()
 
+@register.filter(name='jsonify')
+def json_dumps(data):
+    return json.dumps(data)
+
 @register.assignment_tag
 def assign(val=None):
   return val
 
 @register.filter
 def hash(h, key):
-  if key in h:
-    return h[key]
+  if h:
```

Comment:
hmmm - where was this needed? Do the current templates assume dictionaries can be not defined? Is this not hiding brokenness we're introducing?
Let's at least add a log line - django templates already log for missing context values.

Reply:
Logging added